csFastFloat/FastDoubleParser.cs

```diff
@@ -404,6 +453,55 @@ unsafe static internal double HandleInvalidInput(char* first, char* last)
     }
 
 
+    unsafe static internal double HandleInvalidInput(byte* first, byte* last)
+    {
+      // C# does not (yet) allow literal ASCII strings (it uses UTF-16), so
+      // we need to use byte arrays.
+      // "infinity"  string in ASCII, e.g., 105 = i
+      ReadOnlySpan<byte> infinity_string = new byte[]{105, 110, 102, 105, 110, 105, 116, 121};
+      // "inf" string in ASCII
+      ReadOnlySpan<byte> inf_string = new byte[]{105, 110, 102};
+      // "+inf" string in ASCII
+      ReadOnlySpan<byte> pinf_string = new byte[]{43, 105, 110, 102};
+      // "-inf" string in ASCII
+      ReadOnlySpan<byte> minf_string = new byte[]{5, 105, 110, 102};
+      // "nan" string in ASCII
+      ReadOnlySpan<byte> nan_string = new byte[]{110, 97, 110};
+      // "-nan" string in ASCII
+      ReadOnlySpan<byte> mnan_string = new byte[]{45, 110, 97, 110};
+      // "+nan" string in ASCII
+      ReadOnlySpan<byte> pnan_string = new byte[]{43, 110, 97, 110};
+
+      if (last - first >= 3)
+      {
+        if (Utils.strncasecmp(first, nan_string, 3))
```

Comment:
A potentially different way to solve this, and this is something i've seen by the json parsers is instead of `ReadOnlySpan<byte>`. E.g.

```cs
const uint ninf_uint = 0x666E692D;
var input = Unsafe.Readunaligned<uint>(first)
if ((input &0x20202020) == ninf_uint)
{
   return Double.NegativeInfinity;
}
```

Now, this is the slow and an exceptional path so it may not be that important to consider such optimizations.

Reply:
I suggest to handle it as a separate issue, see https://github.com/CarlVerret/csFastFloat/issues/59